backend/src/test/java/de/neuefische/booklibrary/backend/api/ApiBookIntegrationTest.java

```diff
@@ -0,0 +1,173 @@
+    @Autowired
+    private ObjectMapper objectMapper;
+
+
+    @BeforeAll
+    static void beforeAll() throws IOException {
+        mockWebServer = new MockWebServer();
+        mockWebServer.start();
+    }
+
+    @DynamicPropertySource
+    static void backendProperties(DynamicPropertyRegistry registry) {
+        registry.add("api.url", () -> mockWebServer.url("/").toString());
+
+    }
+
+    @AfterAll
+    static void afterAll() throws IOException {
+        mockWebServer.shutdown();
+    }
+
+
+    @Test
+    void searchApiBooksByIsbn_returnListWithOneBook() throws Exception {
+        //given
+        String previewLink = "http://books.google.de/books/preview";
+        Isbn isbn_13 = new Isbn("ISBN_13", "9783897214484");
+        Isbn isbn_10 = new Isbn("ISBN_10", "3897214482");
+        ImageLinks thumbnail = new ImageLinks("http://books.google.com/books/thumbnail");
+        VolumeInfo volumeInfo = new VolumeInfo("Java von Kopf bis Fuß", List.of("Kathy Sierra", "Bert Bates"), List.of(isbn_10, isbn_13), thumbnail, previewLink);
+        ApiBook book = new ApiBook("5eDWcLzdAcYC", volumeInfo, BookState.AVAILABLE);
+
+        BookResponseElement mockBokListResponse = new BookResponseElement(1, List.of(book));
+        System.out.println("Response:" + mockBokListResponse);
+        mockWebServer.enqueue(new MockResponse()
+                .setBody(objectMapper.writeValueAsString(mockBokListResponse))
+                .addHeader("Content-Type", "application/json")
+        );
+
+
+        //when
+        mockMvc.perform(MockMvcRequestBuilders.get("/api/books/isbn/3897214482"))
+                // then
+                .andExpect(status().isOk())
+                .andExpect(content().json(
+                        """
+                                [
+                                    {
+                                        "id": "5eDWcLzdAcYC",
+                                        "title": "Java von Kopf bis Fuß",
+                                        "author": "Kathy Sierra",
+                                        "isbn": "3897214482",
+                                        "bookState": "AVAILABLE"
+                                    }
+                                ]
+                                """
+                ));
+
+    }
+
+    @Test
+    void searchApiBooksByInvalidIsbn_return0TotalItems() throws Exception {
+        String mockBookListResponse = """
+                 {
+                     "items": [],
+                     "totalItems": 0
+                 }
+                """;
+        mockWebServer.enqueue(new MockResponse()
+                .setBody(mockBookListResponse)
+                .addHeader("Content-Type", "application/json")
+        );
+
+        //when
+        mockMvc.perform(MockMvcRequestBuilders.get("/api/books/isbn/abc"))
+                // then
+                .andExpect(status().isOk())
+                .andExpect(content().json(
```

Comment:
An invalid ISBN returns OK and an empty list? I suppose it is the external API that behaves like that?